lib/master.js

```diff
@@ -35,7 +35,30 @@ class Master extends BaseService {
         this._inRollingRestart = false;
         this._firstWorkerStarted = false;
         this._firstWorkerStartupAttempts = 0;
-        this.workerStatusMap = {};
+        this._workerStatusMap = {};
+        this._shutdownMasterHandler = () => {
+            this.stop()
+            .then(() => {
+                this._logger.log('info/service-runner/master', 'Exiting master');
+                return this._exitProcess(0);
+            });
+        };
+
+        this._rollingRestartHandler = () => {
+            this._firstWorkerStarted = false;
+            this._firstWorkerStartupAttempts = 0;
+            return this._updateConfig()
+            .then(() => {
+                // Recreate loggers
+                this._logger.close();
+                this._logger = new Logger(this.config.logging);
+            })
+            .then(this._rollingRestart.bind(this));
+        };
+
+        this._startupWorkerExitHandler = null;
+        this._workerExitHandler = null;
+        this._workerMessageHandler = null;
```

Comment:
Nonononono... We don't need them to be properties of the master object. Now with many workers we will just override this variable as we start them so it will only remove handlers from the last started worker. These need to be local variable in `_startWorker` function and be confined to it's closure.

Reply:
That's why I had a map there in the first place to for handlers.

Reply:
I understood that. What I'm saying is that you don't need any of those to be properties of a master  object. These all need to be local variables.